### ISSUE TITLE
NullPointerException in ViewGroup code on MoPubFullscreenActivity

### ISSUE THREAD

**Issue by ivan-goncharov-easybrain** [mopub/mopub-android-sdk#472]
`java.lang.NullPointerException: Attempt to read from field 'int android.view.View.mPrivateFlags' on a null object reference`
We noticed a new crash in firebase after we updated MoPub sdk to 5.16.4 (we have no such crash on 5.14).
While stacktrace does not directly point to MoPub SDK we have a reason to believe that it happens in your sdk (more in evidence section).

#### MoPub SDK Version: 
5.16.4
#### Device model and OS Version: 
for now we have 3 crash reports in firebase:
- android 5.1.1 - Lenovo TAB 2 A10-30
- android 11 - Pixel 5
- android 10 - Onn. 10.1
#### Steps to reproduce the behavior:
No steps to reproduce
#### Evidence:
The stacktrace is the following:
```
Caused by java.lang.NullPointerException: Attempt to read from field 'int android.view.View.mPrivateFlags' on a null object reference
       at android.view.ViewGroup.resetCancelNextUpFlag(ViewGroup.java:2188)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2105)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2406)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2107)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2406)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2107)
       at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2406)
       at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2107)
       at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2377)
       at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1727)
       at android.app.Activity.dispatchTouchEvent(Activity.java:2785)
       at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2338)
       at android.view.View.dispatchPointerEvent(View.java:8676)
       at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4141)
       at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4007)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3562)
       at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3615)
       at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3581)
       at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3698)
       at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3589)
       at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3755)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3562)
       at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3615)
       at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3581)
       at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3589)
       at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3562)
       at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:5825)
       at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:5799)
       at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:5770)
       at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:5915)
       at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
       at android.os.MessageQueue.nativePollOnce(MessageQueue.java)
       at android.os.MessageQueue.next(MessageQueue.java:143)
       at android.os.Looper.loop(Looper.java:122)
       at android.app.ActivityThread.main(ActivityThread.java:5268)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:697)
```

The last event in Log section in firebase across all 3 crashes is `screen_view` with `MoPubFullscreenActivity` param. 
![image](https://user-images.githubusercontent.com/56021456/117296536-d7d2e600-ae7d-11eb-91df-1eb412964e95.png)

While doing some research about this crash I've found this article - https://www.jianshu.com/p/ab1fb00fcb90
The conclusion is that this crash can happen if we remove a view during touch event dispatching. Unfortunately I wasn't able to reproduce this bug with provided example.

Looking at the changes in MoPub 5.16.x versions I have a theory that removing views in `FullscreenAdController.onVideoFinish` causes that, since this method is called from `OnTouchListener` in `VastVideoViewController`

**Comment by p158276** [comment]
Hi @ivan-goncharov-easybrain , 
Thanks for bringing that to our attention. For the crash information you provide, 3 crash reports in firebase. Does that all happen within a day, week or month? Are you seeing the same issue with other MoPub SDK versions? We’ll check with the rest of the team on the status, and circle back when there is an update.

Yoyo Lin
MoPub Team

**Comment by ivan-goncharov-easybrain** [comment]
Hi @p158276 
These crashes happened within a week and we found them on the first firebase crashlytics page since the app is not in full rollout yet.  
We don't see this issue with previous MoPub version we used - 5.14.0

**Comment by p158276** [comment]
Hi @ivan-goncharov-easybrain , 
The issue has been fixed in [5.17.0](https://github.com/mopub/mopub-android-sdk/releases/tag/v5.17.0) SDK. Feel free to let us know if you still have any question.

**Comment by ivan-goncharov-easybrain** [comment]
Thanks for a quick resolution!

I'm closing this issue, I'll let you know if the problem will persist.
